src/Utils/localImagesHelpers.ts

```diff
@@ -0,0 +1,149 @@
+import { IMAGES_LOCAL_STORE_LAST_UPDATED_AT } from "Apps/Settings/Routes/MyCollection/constants"
+import localforage from "localforage"
+
+const GEMINI_IMAGE_PROCESS_TIME_IN_MINUTES = 1
+export interface LocalImage {
+  data: string
+  width: number
+  height: number
+}
+
+interface Expirable {
+  expirationDate: string
+}
+
+export type StoredImage = LocalImage & Expirable
+
+export type StoredArtworkWithImages = {
+  artworkID: string
+  images: StoredImage[]
+}
+
+const addMinutes = (date: Date, minutes: number) => {
+  return new Date(date.getTime() + minutes * 60000)
+}
+
+export const storeArtworkLocalImages = async (
```

Comment:
**suggestion:** Have you tried to use the image ID (from Metaphysics) as a key to store the images? Like this, we would be able to fetch the image by its ID later. This could simplify things a lot.

Reply:
I would like to try this out. Can we maybe do this at a follow up ticket since Product wants to get this out of the door asap

Reply:
perfect 👍

Reply:
Great, I will create a ticket now